### DISCUSSION TITLE
Pfd is not loading

### DISCUSSION THREAD

**Discussion by anoop666** [stephanrauh/ngx-extended-pdf-viewer#1613] [Q&A]
Can anyone please help me out
![IMG_20221207_131129.jpg](https://user-images.githubusercontent.com/47858623/206118181-7daefd49-849e-4299-9698-f07dd3864daa.jpg)

**Comment by stephanrauh** [comment]
It's hard to say what's wrong without seeing your source code. My general recommendation is to create a demo project from scratch, just to see if the PDF viewer works on your machine at all. In a nutshell, that's

- `ng new my-great-project`
- `cd my-great-project`
- `ng add ngx-extended-pdf-viewer`
- and adding the demo component to the `app.component.html`.

I've described this in more detail at https://pdfviewer.net/extended-pdf-viewer/getting-started.

Does this work on your machine? Can work find out the difference to your project?

If you're still stuck, please add some source code to your question. Alternatively, start your debugger and tell me about your insights. These days, I'm short of time, so I appreciate every help I can get - including help from users seeking help. So allow me to try to talk you into helping me! :)

Best regards,
Stephan

**Reply by anoop666** [reply]
Hai, 
I have implemented the pdf viewer long ago it was working fine but due to some new  requirements i tried for showing the ngx extended pdf viewer on a popup window and in that case the pdf viewer is not working .
Can we connect  ?

**Reply by stephanrauh** [reply]
Yes, of course. Drop me an email: webmaster1 at beyondjava dot de. However, "popup" is the magic work here. Have a look at https://pdfviewer.net/extended-pdf-viewer/troubleshooting and scroll down to "offsetParent not set". Most likely, your problem is that the PDF viewer isn't visible when it tries to initialize, so it's trying to access DOM elements that aren't there. Hide the PDF viewer by default, and show it with a little timeout after opening the popup.

**Reply by anoop666** [reply]
After putting the time delay also its not working, still have the same issue

**Reply by stephanrauh** [reply]
Ah, the error happens in the `PDFDummyComponentsComponent`. I forgot to add two if-statements. I assume you're using a custom toolbar?

**Reply by anoop666** [reply]
Hai i am not using any custom toolbar , what about the 2 if conditions?..... Is there any way by which we can solve this issue

**Reply by stephanrauh** [reply]
The if conditions are my fault. It's easy to fix, I just need some spare time.

**Reply by anoop666** [reply]
Ok bro i will wait

**Reply by stephanrauh** [reply]
The wait is over! Your bugfix has landed with version 15.2.2. I just wonder what the root cause of your bug is - but chances are an upgrade solves it. Keep me in the loop!

**Reply by anoop666** [reply]
Ok bro let me try

**Reply by anoop666** [reply]
![IMG_20221214_102717.jpg](https://user-images.githubusercontent.com/47858623/207509874-5a9bbe47-550c-4d62-9309-5870e4eb2449.jpg)

**Reply by anoop666** [reply]
> ![IMG_20221214_102717.jpg](https://user-images.githubusercontent.com/47858623/207509874-5a9bbe47-550c-4d62-9309-5870e4eb2449.jpg)

After using the new version i am getting a new error on my console. I will attach the image for your reference also. I hope we can fix this .

**Reply by stephanrauh** [reply]
It seems something basic is wrong in your project. The new error messages indicate that the JavaScript files of pdf.js haven't been loaded or failed to initialize. At the same time, that's nonsense because `onPDFJSInit()` is called by pdf.js itself, when it has finished initializing. I'm puzzled.

**Reply by anoop666** [reply]
But the pop is working quite good with other components but when u tried to use it with the pdf component I don't know and getting errors and due to which my work is not going on 🥲

**Reply by stephanrauh** [reply]
Yeah. I feel you. I'm every bit as puzzled as you are. 

But it's hard to debug an application based on a few screenshots, so I'm clueless.

**Reply by anoop666** [reply]
I have also used another plugin for viewing image and that also works perfectly fine . Even i am clueless

**Reply by anoop666** [reply]
Hai bro, can you please try to call the pdf viewer inside this window tag. This was the code i used for implementing the new window tab popup .
https://stackblitz.com/edit/angular-open-window?file=src%2Fapp%2Fapp.component.ts

Can you try it from your end and if you can get any solution then it will be great.
